flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CompletedCheckpointStore.java

```diff
@@ -84,7 +85,7 @@ default CompletedCheckpoint getLatestCheckpoint(boolean isPreferCheckpointForRec
 	 *
 	 * @param jobStatus Job state on shut down
 	 */
-	void shutdown(JobStatus jobStatus) throws Exception;
+	void shutdown(JobStatus jobStatus, CheckpointsCleaner checkpointsCleaner, Runnable postCleanup) throws Exception;
```

Comment:
Why do we need the `postCleanup`? In the current code base I couldn't find a single usage where we don't pass `() -> {}` @rkhachatryan and @pnowojski? If this is not needed, can we get rid of it? I think it does not make a very nice API with the no-ops. 

Moreover, it would be nice if we could update the JavaDocs if we change the signature of interfaces.

Reply:
`postCleanup` is used by `checkpointsCleaner` but not in case of shutting down. I agree, it should be removed (by pushing down to `DefaultCompletedCheckpointStore`) and javadoc added. My bad.
I've created FLINK-20847 to address this.